src/CoreBundle/private/modules/CMSRunCrons/CMSRunCrons.class.php

```diff
@@ -71,23 +72,36 @@ public function &Execute()
             }
 
             $now = date('Y-m-d');
-            $sQuery = "SELECT * FROM `cms_cronjobs`
-                   WHERE `start_execution` <= '".MySqlLegacySupport::getInstance()->real_escape_string($now)."'
-                     AND (`end_execution` >= '".MySqlLegacySupport::getInstance()->real_escape_string($now)."' || `end_execution` = '0000-00-00')
+            $sQuery = "SELECT `id` FROM `cms_cronjobs`
+                   WHERE `start_execution` <= :now
+                     AND (`end_execution` >= :now || `end_execution` = '0000-00-00')
                      AND `active` = '1'
                 ORDER BY `execute_every_n_minutes`
                  ";
+            $cronJobRows = $this->getDatabaseConnection()->fetchAll($sQuery, ['now' => $now]);
+            foreach ($cronJobRows as $cronJobRow) {
+                $cronJobDb = TdbCmsCronjobs::GetNewInstance();
```

Comment:
-> tdbCronJob

(as it is no database here)

Reply:
renamed